File: miniprogram/utils/cloud-config.js
const SUPABASE_FUNCTIONS_BASE_URL =
  'https://api-test.chuya.wang/functions/v1';
// const SUPABASE_FUNCTIONS_BASE_URL =
  // 'https://rfnjqodcpxxmxinijqcb.supabase.co/functions/v1';

const ENABLE_DEV_LOGIN = false;
const ENABLE_WECHAT_LOGIN = false;
const DEV_LOGIN_KEY = 'farmernote-local-shared-user';
const DEV_LOGIN_DISPLAY_NAME = 'FarmerNote 临时联调';

function isConfigured() {
  return !!SUPABASE_FUNCTIONS_BASE_URL;
}

function isDevLoginEnabled() {
  return ENABLE_DEV_LOGIN;
}

function isWeChatLoginEnabled() {
  return ENABLE_WECHAT_LOGIN;
}

function getDevLoginKey() {
  return DEV_LOGIN_KEY;
}

function getDevLoginDisplayName() {
  return DEV_LOGIN_DISPLAY_NAME;
}

function getFunctionUrl(endpoint) {
  const base = SUPABASE_FUNCTIONS_BASE_URL.replace(/\/+$/, '');
  const cleanedEndpoint = String(endpoint || '').replace(/^\/+/, '');
  return `${base}/${cleanedEndpoint}`;
}

function getStorageOrigin() {
  if (!isConfigured()) {
    return '';
  }

  const match = getFunctionUrl('health').match(/^(https?:\/\/[^/]+)/);
  return match ? match[1] : '';
}

module.exports = {
  getDevLoginDisplayName,
  getDevLoginKey,
  getFunctionUrl,
  getStorageOrigin,
  isDevLoginEnabled,
  isWeChatLoginEnabled,
  isConfigured,
};

File: Flutter/apps/farmernote_app/lib/config/cloud_config.dart
class CloudConfig {
  static const String supabaseFunctionsBaseUrl = String.fromEnvironment(
    'FARMERNOTE_SUPABASE_FUNCTIONS_BASE_URL',
  );
  static const String devLoginMode = String.fromEnvironment(
    'FARMERNOTE_ENABLE_DEV_LOGIN',
    defaultValue: 'false',
  );
  static const String flutterWeChatLoginMode = String.fromEnvironment(
    'FARMERNOTE_ENABLE_FLUTTER_WECHAT_LOGIN',
    defaultValue: 'false',
  );
  static const String devLoginKey = String.fromEnvironment(
    'FARMERNOTE_DEV_LOGIN_KEY',
    defaultValue: 'farmernote-local-shared-user',
  );
  static const String devLoginDisplayName = String.fromEnvironment(
    'FARMERNOTE_DEV_LOGIN_DISPLAY_NAME',
    defaultValue: 'FarmerNote 临时联调',
  );
  static const String flutterWeChatAppId = String.fromEnvironment(
    'FARMERNOTE_FLUTTER_WECHAT_APP_ID',
  );
  static const String flutterWeChatUniversalLink = String.fromEnvironment(
    'FARMERNOTE_FLUTTER_WECHAT_UNIVERSAL_LINK',
  );

  static bool get isSupabaseConfigured => supabaseFunctionsBaseUrl.isNotEmpty;

  static bool get isDevLoginEnabled {
    final normalizedMode = devLoginMode.trim().toLowerCase();
    if (normalizedMode == 'true' || normalizedMode == '1') {
      return true;
    }
    if (normalizedMode == 'false' || normalizedMode == '0') {
      return false;
    }
    return false;
  }

  static bool get isFlutterWeChatConfigured => flutterWeChatAppId.isNotEmpty;

  static bool get isFlutterWeChatLoginEnabled {
    final normalizedMode = flutterWeChatLoginMode.trim().toLowerCase();
    if (normalizedMode == 'true' || normalizedMode == '1') {
      return true;
    }
    if (normalizedMode == 'false' || normalizedMode == '0') {
      return false;
    }
    return false;
  }

  static Uri functionUri(String endpoint) {
    final base = supabaseFunctionsBaseUrl.endsWith('/')
        ? supabaseFunctionsBaseUrl.substring(
            0,
            supabaseFunctionsBaseUrl.length - 1,
          )
        : supabaseFunctionsBaseUrl;
    final cleanedEndpoint = endpoint.startsWith('/')
        ? endpoint.substring(1)
        : endpoint;
    return Uri.parse('$base/$cleanedEndpoint');
  }

  static String get storageOrigin {
    if (!isSupabaseConfigured) {
      return '';
    }
    return functionUri('health').origin;
  }
}
